inference-engine/src/mkldnn_plugin/nodes/mkldnn_convert_node.cpp

```diff
@@ -0,0 +1,92 @@
+// Copyright (C) 2020 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "mkldnn_convert_node.h"
+#include "common/cpu_convert.h"
+
+using namespace mkldnn;
+using namespace MKLDNNPlugin;
+using namespace InferenceEngine;
+
+MKLDNNConvertNode::MKLDNNConvertNode(const InferenceEngine::CNNLayerPtr& layer, const mkldnn::engine& eng, MKLDNNWeightsSharing::Ptr &cache) :
+        MKLDNNNode(layer, eng, cache) {}
+
+void MKLDNNConvertNode::getSupportedDescriptors() {
+    if (outDims.empty() && output.getLayout() != InferenceEngine::Layout::ANY)
+        outDims.push_back(MKLDNNDims(output.getDims()));
+    if (inDims.empty() && input.getLayout() != InferenceEngine::Layout::ANY)
+        inDims.push_back(MKLDNNDims(input.getDims()));
+    if (getParentEdges().size() != 1)
+        THROW_IE_EXCEPTION << "Incorrect number of input edges for layer " << getName();
+    if (getChildEdges().empty())
+        THROW_IE_EXCEPTION << "Incorrect number of output edges for layer " << getName();
+}
+
+void MKLDNNConvertNode::initSupportedPrimitiveDescriptors() {
+    if (!supportedPrimitiveDescriptors.empty())
+        return;
+
+    auto layer = getCnnLayer();
+    if (layer == nullptr) {
+        THROW_IE_EXCEPTION << "Cannot get CNN layer for layer name " << getName();
+    }
+
+    LayerConfig config;
+    DataConfig dataIn;
+    DataConfig dataConfigOut;
+
+    memory::format fmt;
+
+    if (input.getLayout() != InferenceEngine::Layout::ANY && output.getLayout() != InferenceEngine::Layout::ANY) {
+        dataIn.desc = input;
+        config.inConfs.push_back(dataIn);
+
+        const auto layout = config.inConfs[0].desc.getLayout(); // inp/out layouts must be the same
+        dataConfigOut.desc = TensorDesc(output.getPrecision(), input.getDims(), layout);
+        config.outConfs.push_back(dataConfigOut);
+        fmt = MKLDNNMemory::Convert(layout);
+    } else if (layer->insData.size() == 1 && layer->outData.size() == 1) {
+        const SizeVector& ins_dims = layer->insData[0].lock()->getTensorDesc().getDims();
+        const auto layout = layer->insData[0].lock()->getTensorDesc().getLayout(); // inp/out layouts must be the same
+        dataIn.desc = TensorDesc(layer->insData[0].lock()->getTensorDesc().getPrecision(), ins_dims, layout);
+        config.inConfs.push_back(dataIn);
+
+        const SizeVector& out_dims = layer->outData[0]->getTensorDesc().getDims();
+        dataConfigOut.desc = TensorDesc(layer->outData[0]->getTensorDesc().getPrecision(), out_dims, layout);
+        config.outConfs.push_back(dataConfigOut);
+        fmt = MKLDNNMemory::Convert(layout);
```

Comment:
In fact I don't understand why should we create supported pd based on CNNLayer layouts. The best way is to create pds in generalized way for planar/blocked/by_channels layouts (smt similar to SplitNode).

Reply:
A new class, TensorDescCreator, which implements the factory method pattern for creating tensor descriptors, has been added to generalize the initialization of the supported primitive descriptors (see commit e1639f394ed3299b1def7fd04e3484aaeda0f5cd).